src/owl.Router.js

```diff
@@ -26,39 +26,39 @@
         /**
          * Opens page by path
          * @param {string} path Page path
-         * @return {function} Function to destroy controller
+         * @return {Promise<?function>} Function to destroy controller
          */
         open: function(path) {
-            var route = this.getRoute(path);
+            var route = this.getRoute(path), that = this;
             if (!route) {
-                return;
+                return owl.Promise.resolve(null);
             }
 
-            if (this.resolve(route)) {
-                return this.run(path, route);
-            }
-            return null;
+            return this.resolve(route).then(function (ret) {
```

Comment:
Let's name it resolveResult and pass it also this.run and then to action or callback as second parameter.